host_source.go

```diff
@@ -721,3 +732,151 @@ func (r *ringDescriber) refreshRing() error {
 	r.session.policy.SetPartitioner(partitioner)
 	return nil
 }
+
+const (
+	ringRefreshDebounceTime = 1 * time.Second
+)
+
+type ringRefreshDebouncer struct {
+	mu            sync.Mutex
+	stopped       bool
+	broadcaster   *broadcaster
+	timer         *time.Timer
+	refreshNowCh  chan struct{}
+	quit          chan struct{}
+	ringDescriber *ringDescriber
+}
+
+func newRingRefreshDebouncer(r *ringDescriber) *ringRefreshDebouncer {
+	d := &ringRefreshDebouncer{
+		stopped:       false,
+		broadcaster:   nil,
+		refreshNowCh:  make(chan struct{}),
+		quit:          make(chan struct{}),
+		timer:         time.NewTimer(ringRefreshDebounceTime),
+		ringDescriber: r,
+	}
+	d.timer.Stop()
+	go d.flusher()
+	return d
+}
+
+func (d *ringRefreshDebouncer) debounce() {
+	d.timer.Reset(ringRefreshDebounceTime)
```

Comment:
I might add a lock here as well, not sure if concurrent `Reset` and `Stop` calls can be problematic here so should probably lock just in case